iroh/src/endpoint.rs

```diff
@@ -964,28 +980,31 @@ impl Endpoint {
     ///
     /// Returns an error if closing the magic socket failed.
     /// TODO: Document error cases.
-    pub async fn close(&self, error_code: VarInt, reason: &[u8]) -> Result<()> {
-        let Endpoint {
-            msock,
-            endpoint,
-            cancel_token,
-            ..
-        } = self;
-        cancel_token.cancel();
+    pub async fn close_with_code(&self, error_code: VarInt, reason: &[u8]) -> Result<()> {
```

Comment:
I think this should be `close_all_connections(&self, error_code: impl Into<VarInt>, reason: &[u8])` and not actually close all endpoint itself.  That's probably an easier separation to understand.

@matheus23 I can't find your design on this in notion?

Reply:
but it does the same as `.close` in that regard?

Reply:
They're here: https://www.notion.so/number-zero/RFC-draft-Improving-connection-closing-APIs-11d5df1306fb8050b09cf49e521ca741

Reply:
@matheus23 thanks!
@dignifiedquire not sure I understand your question.

I think we need:

- A **sync** `fn close_all_connection(error_code, reason)` that just sends close on all the connections and doesn't wait at all.
- An `async fn close()` as you already implement that closes the endpoint gracefully.

So a recommended way of force-closing everything would be:
```rust
ep.close_all_connection(1, b"shutting down");
ep.close().await?;
```

And of course omitting `close_all_connections` would still work and be well behaved, you only gave up control over the error code and message.

Reply:
Oh, that is a quite different to what I implemented.

I don't understand how this would work though, `close` would still call `magisock::close(0, b"")` so you end up calling close twice in your example above.

Reply:
So I guess we remove this method for now?  Until someone really can't live without?

Reply:
okay will do

Reply:
done